src/Site.js

```diff
@@ -387,6 +409,191 @@ Site.prototype.createPage = function (config) {
+  return fs.copyAsync(filePath, indexPagePath)
+    .catch(() => Promise.reject(new Error(`Failed to copy over ${filePath}`)));
+};
+
+/**
+ * Adds an about page to site if not present.
+ */
+Site.prototype.addAboutPage = function () {
+  const aboutPath = path.join(this.rootPath, ABOUT_MARKDOWN_FILE);
+  return fs.accessAsync(aboutPath)
+    .catch(() => {
+      if (fs.existsSync(aboutPath)) {
+        return Promise.resolve();
+      }
+      return fs.outputFileAsync(aboutPath, ABOUT_MARKDOWN_DEFAULT);
+    });
+};
+
+/**
+ * Adds a footer to default layout of site.
+ */
+Site.prototype.addFooterToDefaultLayout = function () {
+  const footerPath = path.join(this.rootPath, FOOTER_PATH);
+  const siteLayoutPath = path.join(this.rootPath, LAYOUT_FOLDER_PATH);
+  const siteLayoutFooterDefaultPath = path.join(siteLayoutPath, LAYOUT_DEFAULT_NAME, 'footer.md');
+  const wikiFooterPath = path.join(this.rootPath, WIKI_FOOTER_PATH);
+
+  return fs.accessAsync(wikiFooterPath)
+    .then(() => {
+      const footerContent = fs.readFileSync(wikiFooterPath, 'utf8');
+      const wrappedFooterContent = `<footer>\n\t${footerContent}\n</footer>`;
+      return fs.outputFileAsync(siteLayoutFooterDefaultPath, wrappedFooterContent);
+    })
+    .catch(() => {
+      if (fs.existsSync(footerPath)) {
+        return fs.copyAsync(footerPath, siteLayoutFooterDefaultPath);
+      }
+      return Promise.resolve();
+    });
+};
+
+/**
+ * Adds a site navigation bar to the default layout of the site.
+ */
+Site.prototype.addSiteNavToDefaultLayout = function () {
+  const siteLayoutPath = path.join(this.rootPath, LAYOUT_FOLDER_PATH);
+  const siteLayoutSiteNavDefaultPath = path.join(siteLayoutPath, LAYOUT_DEFAULT_NAME, 'navigation.md');
+  const wikiSiteNavPath = path.join(this.rootPath, WIKI_SITE_NAV_PATH);
+
+  return this.readSiteConfig()
+    .then(() => this.collectAddressablePages())
+    .then(() => fs.accessAsync(wikiSiteNavPath))
+    .then(() => {
+      const siteNavContent = this.parseSidebarFileContent(fs.readFileSync(wikiSiteNavPath, 'utf8'));
+      const wrappedSiteNavContent = `<navigation>\n${siteNavContent}\n</navigation>`;
+      return fs.outputFileSync(siteLayoutSiteNavDefaultPath, wrappedSiteNavContent);
+    })
+    .catch(() => this.buildSiteNav(siteLayoutSiteNavDefaultPath));
+};
+
+/**
+ * Parses _Sidebar.md to create a MarkBind friendly site navigation menu.
+ * Formats supported: [[Name|URL]] and [Name](URL).
+ */
+Site.prototype.parseSidebarFileContent = function (rawSiteNavContent) {
+  let siteNavContent = '';
+  const doubleSquareBracketsFormatRegex = /\[\[.+\|{0,1}.?\]\]/g;
+  const markdownLinkFormatRegex = /(?:__|[*#])|\[(.*?)\]\(.*?\)/g;
+  const baseUrlSubst = '{{baseUrl}}';
+  if (rawSiteNavContent.match(doubleSquareBracketsFormatRegex)) {
+    rawSiteNavContent.split('\n').forEach((navLine) => {
+      const nameStartIdx = navLine.indexOf('[[');
```

Comment:
There seems to be two different methods employed to search for double square brackets.  I believe regular expression has a way for you to extract out the URL text and URL link itself, so that we don't have to manually use `.indexOf()` and etc. to manually extract them.